src/main/kotlin/com/terraformation/backend/customer/api/UsersController.kt

```diff
@@ -104,6 +99,38 @@ class UsersController(private val userStore: UserStore) {
     userStore.updatePreferences(payload.organizationId, payload.preferences)
     return SimpleSuccessResponsePayload()
   }
+
+  @ApiResponse200
+  @ApiResponse404
+  @GetMapping
+  @Operation(summary = "Gets a user by some criteria, for now only email is available")
+  fun searchUsers(
+      @RequestParam
+      @Schema(description = "The email to use when searching for a user")
+      email: String
+  ): GetUserResponsePayload {
+    val user = userStore.fetchByEmail(email)
+    if (user != null) {
+      return GetUserResponsePayload(UserProfilePayload(user))
+    }
+
+    throw UserNotFoundForEmailException(email)
+  }
+
+  @ApiResponse200
+  @ApiResponse404
+  @GetMapping("/{userId}")
+  @Operation(summary = "Get a user by ID, if they exist, only ordinary users are supported.")
+  fun getUser(
+      @PathVariable("userId") userId: UserId,
+  ): GetUserResponsePayload {
+    val user = userStore.fetchOneById(userId)
```

Comment:
Similar to previous comment: this needs to prevent exposing all users' data to all other users.

Reply:
Adding the wrapper function now, do you think I should also verify that the found user has an `@terraformation.com` email, otherwise throw "not found"?

Reply:
I think we'll probably find it useful to have "get other user who I have permission to view" as a general-purpose endpoint down the road, so I'd say probably not.

Reply:
Also, the permission I am thinking of adding is "canReadTerraformationUser", that way I can apply it to both `fetchOneById` and `fetchByEmail` wrapper functions. Does that sound alright or should I have a separate permission for each one?